STATION TRAIT GAMEMODE: Naval Patrol (No exterior threats, only internal threats)

DrDiasyl opened the pull request:
## About The Pull Request

The opposite version of #83307

This very rare station trait (1 point less rare as Station-Wide Background Checks) does not allow any exterior antagonists like Nuclear Operatives, Wizards, Space Dragons, Aliens, etc to spawn, the only exception is Blob Infection.

Internal threats like Traitors, Changelings, Heretic, Cults, Revolutionaries, etc can still spawn while the Station Trait is active.
## Why It's Good For The Game
When #83307 only appeared, I immediately thought it would be neat to have the opposite: only internal threats and no outside threats.
And after the success of #83307, this will work out well. Especially that more "Station Trait Gamemodes" would be neat to have.

This would allow the Station to be fully submerged only in internal paranoia threats without being at risk of getting all of that atmosphere broken by a sudden Space Dragon or similar that everyone must murder and abandon whatever they did before.
Rare rounds with this threat would be refreshing for Crew and Security as they have no stress of getting randomly under heavy fire in the middle of the round, without just being a Green Start/White Dwarf with 0 threats.

Paranoia and investigations will be more plausible when the Station Trait is active.
## Changelog
:cl: DrDiasyl aka DrTuxedo
add: NEW STATION TRAIT "GAMEMODE": Naval Patrol! The station sector is being crossed by a Naval fleet patrol, due to this NO external threats or infestations can board the station. But internal threats remain, so stay vigilant and paranoid without getting distracted...
/:cl:

DrDiasyl committed:
Naval Patrol

DrDiasyl committed:
Update neutral_traits.dm

Loafin34 commented:
The good old days of fixed gamemodes..

Ghommie commented:
Finally, you can tuck the nuke disk into bed without worrying about evil operatives showing up. Sleep tight, diskie!

DrDiasyl commented:
> Finally, you can tuck the nuke disk into bed without worrying about evil operatives showing up. Sleep tight, diskie!

Well I didn't find Lone Ops in the dynamic rulesets (because probably its not a one), so I think Lone Ops are still possible if Captain is sloppy and left the disk.

Ghommie commented:
Oh noes...

DrDiasyl committed:
It alerts about lone ops

Its funny.

Lone Ops are easily avoided by just taking the disk - if you are not taking the disk then even Navy can't do anything.

At least they will tell you about this.

CCC23 commented:
the true lone op was in the station the WHOLE TIME

Ghommie commented on code/datums/station_traits/neutral_traits.dm:
```diff
@@ -544,3 +544,16 @@
+	can_revert = FALSE
+
+	dynamic_category = RULESET_CATEGORY_NO_OUTSIDE_ANTAGONISTS
+	threat_reduction = 15
```
I wonder if reducing the threat is a good thing in this case, but whatever

Ghommie reviewed (changes requested):
All good, but you need to make make the naval patrol and background_checks traits not override each other. Check the code for `dynamic_category`, you just have to change the `=` operator with `|=` in one place and then add a line to remove `RULESET_CATEGORY_DEFAULT` from the categories bitfield.

Alternatively you could make them blacklist each other but that would only hide the underlying issue.

carlarctg commented:
I had a similar concept, but it would be 'dynamic themes' and they would only rarely be announced to the crew. Such as only magic users, only syndicate, only outside threats... this works too though.

This is going to be way too rare if it's as common as 'no antags'. Please up the likelihood or it'll barely even exist.

SomeoneYouProbablyKnow commented:
Does this meaningfully change the round? Knowing i can't trust the other crewmen fully is a pretty natural state of the game. It's not much different from a black orbit where you really don't wanna trust anyone. That's why the background check thing works so well because it subverts that natural state.

Hardly any of my decisions are influenced by the knowledge that a nuckie team MAY spawn
And not like this limites the round to small scale antags only. revs ascended heretics cult all need you to get up from your spor and do stuff about em

DrDiasyl commented:
> Does this meaningfully change the round? Knowing i can't trust the other crewmen fully is a pretty natural state of the game. It's not much different from a black orbit where you really don't wanna trust anyone. That's why the background check thing works so well because it subverts that natural state.

Well, it is not the biggest deviation, but it still makes it so the game is 100% paranoia-focused with no "Oops, all dragons" in the middle of the round as explained in the "Why It's Good For Game" section.
Might increase its chance of appearing to be honest.

> Hardly any of my decisions are influenced by the knowledge that a nuckie team MAY spawn

But if they do spawn (or any other big midround) all of the paranoia falls as now everyone must be focused on this one specific threat ignoring the rest.

> And not like this limites the round to small scale antags only. revs ascended heretics cult all need you to get up from your spor and do stuff about em

Those are still internal threats, coming from the crew itself which start stealthy and can be snooped out early. Something you can't do with exterior threats as those appear suddenly.

SomeoneYouProbablyKnow commented:
> Well, it is not the biggest deviation, but it still makes it so the game is 100% paranoia-focused with no "Oops, all dragons" 

> But if they do spawn (or any other big midround) all of the paranoia falls as now everyone must be focused on this one specific threat ignoring the rest.

But the same thing happens with revs, cult and to a lesser extent ascended heretics. Someone yelling "cult med maints" breaks all sense of paranoia the same way "nuckies cargo" does. The mechanics of cult and revs don't create paranoia because they'll just be bashing your head in and you'll have to arm up exactly like wtih a dragon. it's a shame cult doesn't have you wondering "is my coworker in a cult" because if they were they'd be making claymores in their base.

If you want a trait that zeros in onnthe paranoia cult and revs shouldn't be there. If they are left there the trait won't change anything for the feel of the round.
Kinda why they were their own gamemodes before dynamic like nuckies were (i'm like 99% sure that's true but double check me)

DrDiasyl commented:
> 
> But the same thing happens with revs, cult and to a lesser extent ascended heretics. Someone yelling "cult med maints" breaks all sense of paranoia the same way "nuckies cargo" does. The mechanics of cult and revs don't create paranoia because they'll just be bashing your head in and you'll have to arm up exactly like wtih a dragon. it's a shame cult doesn't have you wondering "is my coworker in a cult" because if they were they'd be making claymores in their base.
> 
> If you want a trait that zeros in onnthe paranoia cult and revs shouldn't be there. If they are left there the trait won't change anything for the feel of the round.
> Kinda why they were their own gamemodes before dynamic like nuckies were (i'm like 99% sure that's true but double check me)

I don't think Cult or Rev spawns that often to be a constant concern about in this Station Trait.

SomeoneYouProbablyKnow commented:
> I don't think Cult or Rev spawns that often to be a constant concern about in this Station Trait.

They probably spawn about as often as nuckies and wizards give or take, if not more often. You'll end up with a trait that makes for normal rounds but all the potential nuckie rounds are cult or revs rounds. It's not often, no, but then it's a dead trait that doesn't do much for the feel of the round. It doesn't flip the script like backround checks is my point. You end up with the same distribution of "paranoia" antags and "emergency" antags.

 There is perhaps a psychological effect to it. If players are told not to trust each other they won't trust each other. But... as just a trait and a single announcement, i don't expect that to be too strong.

DrDiasyl committed:
ghom suggestions

unit0016 commented on code/modules/events/ghost_role/operative.dm:
```diff
@@ -32,6 +32,13 @@
+	// You should've secured that disk! Not even Navy will stop a disk-sniffing-determined Lone Operative
+	if(GLOB.dynamic_ruleset_categories == RULESET_CATEGORY_NO_OUTSIDE_ANTAGONISTS)
+		minor_announce(
+			message = "We've detected a fast-moving, human-sized, blood-red blip on radar heading towards [GLOB.station_name]. It somehow evaded our engagement protocols and is now within your proximity. The situation is yours to handle, no reply.",
```
This is a neat detail -- can we get the same for the battlecruiser final objective? (not that they don't announce themselves explosively 5 minutes after spawn regardless)

ElGitificador commented:
sounds like this trait would end up in a lot of threat not being spent, since "exterior" midrounds tend to be way more heavy weight wise than the "interior" ones

Ghommie commented on code/datums/station_traits/_station_trait.dm:
```diff
@@ -48,7 +48,7 @@ GLOBAL_LIST_EMPTY(lobby_station_traits)
 		GLOB.dynamic_station_traits[src] = threat_reduction
 	if(dynamic_category)
-		GLOB.dynamic_ruleset_categories = dynamic_category
+		GLOB.dynamic_ruleset_categories |= dynamic_category
```
This won't do, because the global bitfield will still have the `RULESET_CATEGORY_DEFAULT` bitflag enabled by default, and thus the if statement will pass even for rulesets without the added flag:
https://github.com/tgstation/tgstation/blob/54068c2de574a2d9b25759d5f6cd89b56da26ef8/code/controllers/subsystem/dynamic/dynamic.dm#L946

JupiterJaeden commented:
Ignoring the balance/gameplay implications, I do think this is just really fun from a flavor perspective. Very flavorful.
